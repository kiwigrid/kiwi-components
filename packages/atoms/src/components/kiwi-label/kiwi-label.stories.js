import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Label',
  parameters: { docs: { description: { component: notes } } },
  decorators: [],
};

export const basic = (args) => {
  const { kind, classes, label } = args;

  return html` <div class="panel panel-default">
    <div class="panel-body">
      <kiwi-label kind=${kind} classes=${classes}>${label}</kiwi-label>
    </div>
  </div>`;
};
basic.args = {
  classes: '',
  label: 'Click Me!',
};
basic.argTypes = {
  kind: {
    control: {
      type: 'select',
      options: ['default', 'info', 'success', 'warning', 'danger'],
    },
    defaultValue: 'default',
  },
};
