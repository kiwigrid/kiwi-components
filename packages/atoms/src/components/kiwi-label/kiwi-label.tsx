import { Component, ComponentInterface, h, JSX, Prop } from '@stencil/core';
import { invert, stubTrue, mapValues } from 'lodash-es';

export type LabelKind = 'default' | 'info' | 'success' | 'warning' | 'danger';

const LABEL_CLASSES: { [Key in LabelKind]: string } = {
  default: 'label-default',
  info: 'label-info',
  success: 'label-success',
  warning: 'label-warning',
  danger: 'label-danger',
};

@Component({
  tag: 'kiwi-label',
})
export class KiwiLabel implements ComponentInterface {
  /** Kind of label */
  @Prop()
  kind!: LabelKind;

  /** Additional classes to apply to the label. */
  @Prop()
  classes: string[] = [];

  render(): JSX.Element {
    const labelClass = LABEL_CLASSES[this.kind] || LABEL_CLASSES['default'];
    const classes = mapValues(
      invert([...this.classes, labelClass, 'label']),
      stubTrue,
    );
    return (
      <span class={classes}>
        <slot></slot>
      </span>
    );
  }
}
