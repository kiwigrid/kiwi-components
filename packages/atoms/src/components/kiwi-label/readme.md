# kiwi-label



<!-- Auto Generated Below -->


## Properties

| Property            | Attribute | Description                               | Type                                                        | Default     |
| ------------------- | --------- | ----------------------------------------- | ----------------------------------------------------------- | ----------- |
| `classes`           | --        | Additional classes to apply to the label. | `string[]`                                                  | `[]`        |
| `kind` _(required)_ | `kind`    | Kind of label                             | `"danger" \| "default" \| "info" \| "success" \| "warning"` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
