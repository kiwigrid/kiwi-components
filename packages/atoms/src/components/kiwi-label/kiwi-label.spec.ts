import { KiwiLabel, LabelKind } from './kiwi-label';
import { toPairs } from 'lodash-es';
import { newSpecPage } from '@stencil/core/testing';

const testCases: { [Key in LabelKind]: Key } = {
  danger: 'danger',
  default: 'default',
  info: 'info',
  success: 'success',
  warning: 'warning',
};

describe('kiwi-label', () => {
  it('builds', () => {
    expect(new KiwiLabel()).toBeTruthy();
  });

  it.each(toPairs(testCases))('renders %s', async (name, kind) => {
    const label = await newSpecPage({
      components: [KiwiLabel],
      html: `
        <kiwi-label kind="${kind}">${name}</kiwi-label>
      `,
    });

    expect(label.root).toMatchSnapshot();
  });
});
