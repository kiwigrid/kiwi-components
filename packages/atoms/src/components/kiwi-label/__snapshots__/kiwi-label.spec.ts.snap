// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-label renders danger 1`] = `
<kiwi-label kind="danger">
  <!---->
  <span class="label label-danger">
    danger
  </span>
</kiwi-label>
`;

exports[`kiwi-label renders default 1`] = `
<kiwi-label kind="default">
  <!---->
  <span class="label label-default">
    default
  </span>
</kiwi-label>
`;

exports[`kiwi-label renders info 1`] = `
<kiwi-label kind="info">
  <!---->
  <span class="label label-info">
    info
  </span>
</kiwi-label>
`;

exports[`kiwi-label renders success 1`] = `
<kiwi-label kind="success">
  <!---->
  <span class="label label-success">
    success
  </span>
</kiwi-label>
`;

exports[`kiwi-label renders warning 1`] = `
<kiwi-label kind="warning">
  <!---->
  <span class="label label-warning">
    warning
  </span>
</kiwi-label>
`;
